src/components/dropdownSearch/DropdownSearch.tsx

```diff
@@ -262,10 +261,16 @@ export class DropdownSearch extends React.Component<IDropdownSearchProps, void>
       this.props.onKeyDownFilterBox(e.keyCode);
     }
 
-    if (this.props.onOptionClickCallBack && (e.keyCode === keyCode.enter || e.keyCode === keyCode.tab) && this.props.activeOption) {
-      this.props.onOptionClickCallBack(this.props.activeOption);
+    if (e.keyCode === keyCode.enter || e.keyCode === keyCode.tab) {
+      // prevent onClick because an enter key on an input trigger an click event. OnClick re open the dropdown.
+      // prevent the tab event to select the option and change the focus on next element.
+      // We only want to close the dropdown and get focus on him.
```

Comment:
// Prevents the onClick event to be triggered by the Enter key, which would erroneously re-open the dropdown.
// Prevents default events on Tab key to avoid selecting the next option.
// We only want to close the dropdown and display the clicked option as the selected option on these events.